packages/markdown-docx/src/ToCiceroMarkVisitor.js

```diff
@@ -270,24 +286,114 @@ class ToCiceroMarkVisitor {
                     rootNode.nodes = [...rootNode.nodes, constructedNode];
                     rootNodesLength++;
                 } else if (commonPropertiesLength === 1) {
-                    rootNode.nodes[rootNodesLength - 1].nodes = [
-                        ...rootNode.nodes[rootNodesLength - 1].nodes,
-                        constructedNode,
-                    ];
+                    if (propertiesCurrent[commonPropertiesLength - 1] === TRANSFORMED_NODES.optional) {
+                        if (this.JSONXML[nodeIndex].whenSomeType) {
+                            rootNode.nodes[rootNodesLength - 1].whenSome = [
+                                ...rootNode.nodes[rootNodesLength - 1].whenSome,
+                                constructedNode,
+                            ];
+                        } else {
+                            rootNode.nodes[rootNodesLength - 1].whenNone = [
+                                ...rootNode.nodes[rootNodesLength - 1].whenNone,
+                                constructedNode,
+                            ];
+                        }
+                        if (rootNode.nodes[rootNodesLength - 1].hasSome) {
+                            rootNode.nodes[rootNodesLength - 1].nodes = rootNode.nodes[rootNodesLength - 1].whenSome;
+                        } else {
+                            rootNode.nodes[rootNodesLength - 1].nodes = rootNode.nodes[rootNodesLength - 1].whenNone;
+                        }
+                    } else {
+                        rootNode.nodes[rootNodesLength - 1].nodes = [
+                            ...rootNode.nodes[rootNodesLength - 1].nodes,
+                            constructedNode,
+                        ];
+                    }
```

Comment:
This and some chunks below it more or less look similar. You increase the level of nesting for each of them on observation. Maybe create a generic function for them which takes in `rootNode` and `level` and recurses `level` number of times to return the desired nodes.